WeScan/Scan/ScannerViewController.swift

```diff
@@ -177,19 +180,98 @@ final class ScannerViewController: UIViewController {
     
     // MARK: - Actions
     
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesBegan(touches, with: event)
+        
+        guard !CaptureSession.current.isAutoModeEnabled,
+            let touch = touches.first else { return }
+        let touchPoint = touch.location(in: view)
+        let focusPoint: CGPoint = videoPreviewLayer.captureDevicePointConverted(fromLayerPoint: touchPoint)
+        
+        if let focusRectangle = focusRectangle {
+            focusRectangle.removeFromSuperview()
+        }
+        
+        focusRectangle = FocusRectangle(touchPoint: touchPoint)
+        view.addSubview(focusRectangle)
+        
+        defer {
+            if let focusRectangle = focusRectangle {
+                UIView.animate(withDuration: 0.3, delay: 1.0, animations: {
+                    focusRectangle.alpha = 0.0
+                }, completion: { (_) in
+                    focusRectangle.removeFromSuperview()
+                })
+            }
+        }
+        
+        do {
+            guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
+            try device.lockForConfiguration()
+            
+            if device.isFocusPointOfInterestSupported && device.isFocusModeSupported(.autoFocus) {
+                device.focusPointOfInterest = focusPoint
+                device.focusMode = .autoFocus
+            }
+
+            if device.isExposurePointOfInterestSupported, device.isExposureModeSupported(.continuousAutoExposure) {
+                device.exposurePointOfInterest = focusPoint
+                device.exposureMode = .continuousAutoExposure
+            }
+            
+            device.unlockForConfiguration()
+        } catch {
+            let error = ImageScannerControllerError.inputDevice
+            guard let captureSessionManager = captureSessionManager else { return }
+            captureSessionManager.delegate?.captureSessionManager(captureSessionManager, didFailWithError: error)
+            return
+        }
+        
```

Comment:
I think this function is getting a bit long, it would be nice to split this up into smaller functions that this one calls. Preferably with nice names so we can understand the steps taken.

Reply:
Resolved: 9d1e1e4